fastmri/pl_modules/data_module.py

```diff
@@ -21,14 +21,40 @@ def worker_init_fn(worker_id):
     data: Union[
         SliceDataset, CombinedSliceDataset
     ] = worker_info.dataset  # pylint: disable=no-member
+
+    # Check if we are using DDP
+    is_ddp = False
+    if torch.distributed.is_available():
+        if torch.distributed.is_initialized():
+            is_ddp = True
+
     # for NumPy random seed we need it to be in this range
-    seed = worker_info.seed % (2 ** 32 - 1)  # pylint: disable=no-member
+    base_seed = worker_info.seed % (2 ** 32 - 1)  # pylint: disable=no-member
 
     if isinstance(data, CombinedSliceDataset):
         for i, dataset in enumerate(data.datasets):
             if dataset.transform.mask_func is not None:
-                dataset.transform.mask_func.rng.seed((seed + i) % (2 ** 32 - 1))
+                if (
+                    is_ddp
+                ):  # DDP training: unique seed is determined by worker, device, dataset
+                    seed_i = (
+                        base_seed
+                        - worker_info.id
+                        + torch.distributed.get_rank()
+                        * (worker_info.num_workers * len(data.datasets))
+                        + worker_info.id * len(data.datasets)
+                        + i
+                    ) % (2 ** 32 - 1)
+                else:
+                    seed_i = (base_seed + i) % (2 ** 32 - 1)
```

Comment:
I think one more change is needed here. `seed_i = (base_seed + worker_info.id * len(data.datasets) + i) % (2 ** 32 -1)`.

Reply:
Good point. Shouldn't it be `(base_seed - worker_info.id + worker_info.id * len(data.datasets) + i) % (2 ** 32 -1)`, since `base_seed` already has the worker id added to it internally?

Reply:
You're right!